app/org/sagebionetworks/bridge/models/schedules/Activity.java

```diff
@@ -1,49 +1,132 @@
 package org.sagebionetworks.bridge.models.schedules;
 
+import static org.sagebionetworks.bridge.models.schedules.ActivityType.SURVEY;
+import static org.sagebionetworks.bridge.models.schedules.ActivityType.TASK;
+
 import java.util.Objects;
 
+import org.joda.time.DateTime;
+import org.sagebionetworks.bridge.models.BridgeEntity;
+import org.sagebionetworks.bridge.validators.ActivityValidator;
+import org.sagebionetworks.bridge.validators.Validate;
+
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
-public class Activity {
-
-    private final String label;
-    private final ActivityType activityType;
-    private final String ref;
+/**
+ * An activity we wish a study participant to do. The two main types of activities are 
+ * tasks in the application (such as taking a tapping test), and surveys to be filled 
+ * out and returned to the sever.
+ * 
+ * In schedules, activities are set with surveys. In tasks, these are "resolved" to point 
+ * to a specific survey, and a survey response object is also added to the activity. This includes 
+ * an endpoint where answers to the survey can be submitted.
+ */
+public final class Activity implements BridgeEntity {
     
+    private static final ActivityValidator VALIDATOR = new ActivityValidator();
+    
+    private String label;
+    private String labelDetail;
+    private TaskReference task;
+    private SurveyReference survey;
+    private SurveyResponseReference response;
+    private ActivityType activityType;
+
     @JsonCreator
-    public Activity(@JsonProperty("label") String label, @JsonProperty("ref") String ref) {
+    private Activity(@JsonProperty("label") String label, @JsonProperty("labelDetail") String labelDetail, 
+        @JsonProperty("task") TaskReference task, @JsonProperty("survey") SurveyReference survey, 
+        @JsonProperty("surveyResponse") SurveyResponseReference response) {
         this.label = label;
-        this.ref = ref;
-        if (ref == null) {
-            this.activityType = null;
-        } else {
-            this.activityType = SurveyReference.isSurveyRef(ref) ? ActivityType.SURVEY : ActivityType.TASK;    
-        }
+        this.labelDetail = labelDetail;
+        this.survey = survey;
+        this.task = task;
+        this.survey = survey;
+        this.response = response;
+        this.activityType = (task != null) ? TASK : SURVEY;
```

Comment:
Composition over inheritance. ;-)

Actually I started with the JSON I wanted and it suggested sub-objects, but I'm sure Jackson would allow me to express sub-type fields as nested object properties without actually having the objects. Something to keep in mind.